federal-reporting-service/FederalReportingController.cs

```diff
@@ -257,5 +267,31 @@ private string GetDocumentLocationReferenceByRelativeURL(string relativeUrl, str
 
             return result;
         }
+
+        private bool CreateFolder(string folderName)
+        {
+            try
+            {
+                
+                var createFolderRequest = new CreateFolderRequest()
+                {
+                    EntityName = "federal_report",
+                    FolderName = folderName
+                };
+
+                var createFolderResult = _fileManagerClient.CreateFolder(createFolderRequest);
+
+                if (createFolderResult.ResultStatus == ResultStatus.Fail)
+                {
+                    _logger.LogError($"Error creating folder for entity federal_report and folder {folderName}. Error is {createFolderResult.ErrorDetail}");
```

Comment:
We need to sanitize the folder name before putting it in the logs.  See file controller in the main app for an example.